Makefile

```diff
@@ -62,13 +62,50 @@ draw_deps:
 ########################################
 ### Testing
 
-test:
-	@echo "--> Running go test"
-	@go test $(PACKAGES)
+## required to be run first by most tests
+build_docker_test_image:
+	docker build -t tester -f ./test/docker/Dockerfile .
+
+### coverage, app, persistence, and libs tests
+test_cover:
+	# cleanup
+	bash ./test/test.sh
+	# run the go unit tests with coverage (in docker)
+	docker run --name run_test -t tester bash test/test_cover.sh
+	
+test_cover_ci:
```

Comment:
the `*_ci` suffix is temporary and replaces some logic in the `test.sh` file. The won't be necessary when jenkins is setup